src/mappings/twab-controller.ts

```diff
@@ -4,88 +4,118 @@ import {
-import { createTwab } from '../helpers/createTwab';
+import { createTwabObservation, setTwabObservation } from '../helpers/twabObservation';
+import { createBalanceChange, setBalanceChange } from '../helpers/balanceChange';
 import { loadOrCreateAccount } from '../helpers/loadOrCreateAccount';
 import { loadOrCreateVault } from '../helpers/loadOrCreateVault';
-import { setTwab } from '../helpers/twab';
+import { createVaultSupplyChange, setVaultSupplyChange } from '../helpers/vaultSupplyChange';
+import { createVaultTwabObservation, setVaultTwabObservation } from '../helpers/vaultTwabObservation';
 
 export function handleIncreasedBalance(event: IncreasedBalance): void {
-    const vault = event.params.vault;
-    const user = event.params.user;
-    const amount = event.params.amount;
-    const delegateAmount = event.params.delegateAmount;
-    const isNew = event.params.isNew;
+    const { vault, user, amount, delegateAmount } = event.params;
 
     const account = loadOrCreateAccount(vault, user);
     increaseBalance(account, amount);
     increaseDelegateBalance(account, delegateAmount);
 
     const timestamp = event.block.timestamp;
-    const twab = createTwab(vault, user, timestamp);
-    setTwab(twab, account, amount, delegateAmount, isNew, timestamp);
+    const balanceChange = createBalanceChange(vault, user, timestamp);
+    setBalanceChange(balanceChange, account, amount, delegateAmount, timestamp);
 
-    twab.save();
+    balanceChange.save();
     account.save();
 }
 
 export function handleDecreasedBalance(event: DecreasedBalance): void {
-    const vault = event.params.vault;
-    const user = event.params.user;
-    const amount = event.params.amount;
-    const delegateAmount = event.params.delegateAmount;
-    const isNew = event.params.isNew;
+    const { vault, user, amount, delegateAmount } = event.params;
 
     const account = loadOrCreateAccount(vault, user);
     decreaseBalance(account, amount);
     decreaseDelegateBalance(account, delegateAmount);
 
     const timestamp = event.block.timestamp;
-    const twab = createTwab(vault, user, timestamp);
-    setTwab(twab, account, amount, delegateAmount, isNew, timestamp);
+    const balanceChange = createBalanceChange(vault, user, timestamp);
+    setBalanceChange(balanceChange, account, amount.neg(), delegateAmount.neg(), timestamp);
 
-    twab.save();
+    balanceChange.save();
     account.save();
 }
 
 export function handleIncreasedTotalSupply(event: IncreasedTotalSupply): void {
-    const vault = event.params.vault;
-    const amount = event.params.amount;
+    const { vault, amount } = event.params;
 
     const vaultEntity = loadOrCreateVault(vault);
     increaseTotalSupply(vaultEntity, amount);
 
+    const timestamp = event.block.timestamp;
+    const supplyChange = createVaultSupplyChange(vault, timestamp);
+    setVaultSupplyChange(supplyChange, vaultEntity, amount, timestamp);
```

Comment:
Resolved with 4ad6d46

Abstracted these creations / updates (except for the delegation 
handler since it is code that is only used once and so there's no need to abstract it)